Python/Product/Profiling/Profiling/CommandBuilder.cs

```diff
@@ -0,0 +1,170 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+
+
+namespace Microsoft.PythonTools.Profiling {
+    using System;
+    using System.Diagnostics;
+    using System.IO;
+    using System.Linq;
+    using System.Windows;
+    using Microsoft.PythonTools.Infrastructure;
+    using Microsoft.PythonTools.Interpreter;
+
+    internal class CommandBuilder {
+
+        /// <summary>
+        /// Constructs a <see cref="TargetCommand"/> based on the provided profiling target.
+        /// </summary>
+        public TargetCommand BuildCommandFromTarget(ProfilingTarget target) {
+            try {
+                if (target == null) {
+                    return null;
+                }
+
+                var pythonProfilingPackage = PythonProfilingPackage.Instance;
+                var joinableTaskFactory = pythonProfilingPackage.JoinableTaskFactory;
+
+                TargetCommand command = null;
+
+                joinableTaskFactory.Run(async () => {
+                    await joinableTaskFactory.SwitchToMainThreadAsync();
+
+                    var name = target.GetProfilingName(pythonProfilingPackage, out var save);
```

Comment:
Why does this need to be on the main thread?

Reply:
The `ShowPerformanceExplorerAsync()` shows a UI element, and from what I remember, any UI related operations need to happen on the main thread(UI thread).

Reply:
Ah that makes sense. Thanks.